app/code/Magento/QuoteGraphQl/Model/Resolver/CartItemTypeResolverComposite.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteGraphQl\Model\Resolver;
+
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\TypeResolverInterface;
+
+/**
+ * {@inheritdoc}
+ */
+class CartItemTypeResolverComposite implements TypeResolverInterface
+{
+    /**
+     * TypeResolverInterface[]
+     */
+    private $cartItemTypeResolvers = [];
+
+    /**
+     * @param TypeResolverInterface[] $cartItemTypeResolvers
+     */
+    public function __construct(array $cartItemTypeResolvers = [])
+    {
+        $this->cartItemTypeResolvers = $cartItemTypeResolvers;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws GraphQlInputException
+     */
+    public function resolveType(array $data) : string
+    {
+        if (!isset($data['product'])) {
+            throw new GraphQlInputException(
+                __('Missing key %1 in cart data', ['product'])
```

Comment:
I don't think this error will be useful for the end user. This should probably be a regular `\LogicException`. What do you think?